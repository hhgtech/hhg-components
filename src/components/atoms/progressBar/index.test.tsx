import React from 'react'
import { render, cleanup } from '@testing-library/react'
import 'jest-styled-components'

import { ProgressBar } from '.'

import MockTheme from './../../../mockTheme'

afterEach(cleanup)

test('it renders', () => {
  const progressBar = render(
    <MockTheme>
      <ProgressBar
        numberOfStep={4}
        labelPrefix="Step"
        activeStep={2}
        className="custom-class"
        onChange={(step) => {
          console.log(step)
        }}
      />
    </MockTheme>,
  )

  const self = progressBar.container.firstChild
  expect(self).toMatchSnapshot()
})
