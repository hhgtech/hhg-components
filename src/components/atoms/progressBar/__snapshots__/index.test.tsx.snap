// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
.c0 {
  margin: 0 12px;
  position: relative;
}

.c0 div[role='slider'] {
  -webkit-text-size-adjust: none;
  -webkit-user-select: none;
}

.c0 div[role='slider']:focus {
  outline: none;
}

.c3 {
  position: absolute;
  top: 19px;
  width: 24px;
  height: 24px;
  color: #fff;
  border: 2px solid #fff;
  border-radius: 50%;
  -webkit-transform: translateY(-50%) translateX(-50%);
  -ms-transform: translateY(-50%) translateX(-50%);
  transform: translateY(-50%) translateX(-50%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  background-color: #fff;
  color: #bfbfbf;
  border: 1px solid #bfbfbf;
}

.c5 {
  position: absolute;
  top: 19px;
  width: 24px;
  height: 24px;
  color: #fff;
  border: 2px solid #fff;
  border-radius: 50%;
  -webkit-transform: translateY(-50%) translateX(-50%);
  -ms-transform: translateY(-50%) translateX(-50%);
  transform: translateY(-50%) translateX(-50%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  background-color: #2d87f3;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 36px;
  margin-top: 32px;
}

.c2 {
  width: 100%;
  height: 5px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  border-radius: 4px;
  background-color: #d9d9d9;
}

.c4 {
  position: absolute;
  top: 32px;
  min-width: 100px;
  text-align: center;
  color: #262626;
  font-size: 13px;
}

<div
  class="c0 custom-class"
>
  <div
    class="c1 progress-track"
  >
    <div
      class="c2"
    />
  </div>
  <div
    class="c3"
    data-is-active="false"
    data-label="0"
    style="left: 0px;"
  >
    1
    <div
      class="c4"
    >
      Step 1
    </div>
  </div>
  <div
    class="c3"
    data-is-active="false"
    data-label="1"
    style="left: 0px;"
  >
    2
    <div
      class="c4"
    >
      Step 2
    </div>
  </div>
  <div
    class="c5"
    data-is-active="true"
    data-label="2"
    style="left: 0px;"
  >
    3
    <div
      class="c4"
    >
      Step 3
    </div>
  </div>
  <div
    class="c3"
    data-is-active="false"
    data-label="3"
    style="left: 0px;"
  >
    4
    <div
      class="c4"
    >
      Step 4
    </div>
  </div>
</div>
`;
